File: gemini/db_libs.py
bid_ask_create_str = '''
CREATE TABLE gemini_{}_bid_ask (
	ts 			timestamp without time zone, 	-- date of insert
	symbol 		varchar(6),						-- symbol for currency
	type 		boolean,						-- false -> bid; true -> ask
	price 		real,							-- currency price
	remaining 	real
)
;
'''

trade_create_str = '''
CREATE TABLE gemini_{}_trade (
	ts 			timestamp without time zone,	-- date of insert
	symbol		varchar(6),						-- symbol for currency
	type		boolean,						-- false -> bid; true -> ask
	price		real,							-- execution price
	amount		real							-- trade quantity
)
;
'''


bid_ask_drop_str = '''
DROP TABLE gemini_{}_bid_ask
;
'''

trade_drop_str = '''
DROP TABLE gemini_{}_trade
;
'''


bid_ask_insert_str = '''
INSERT INTO gemini_{}_bid_ask (
	ts,
	symbol,
	type,
	price,
	remaining
) VALUES (
	CURRENT_TIMESTAMP, '{}', {}, {}, {}
)
;
'''

trade_insert_str = '''
INSERT INTO gemini_{}_trade (
	ts,
	symbol,
	type,
	price,
	amount
) VALUES (
	CURRENT_TIMESTAMP, '{}', {}, {}, {}
)
;
'''


if __name__ == '__main__':
	import os
	import psycopg2
	import sys

	symbol = 'btc'

	connect_dsn = "dbname='crypto' user='{}' host='localhost' password='{}'".format(os.environ.get('DBUSER'), os.environ.get('DBPWD'))

	try:
		conn = psycopg2.connect(connect_dsn)
	except:
		print('unable to create db connection')

	cur = conn.cursor()

	try:
		cur.execute(bid_ask_drop_str.format(symbol))
		cur.execute(bid_ask_create_str.format(symbol))

		cur.execute(trade_drop_str.format(symbol))
		cur.execute(trade_create_str.format(symbol))

		conn.commit()
		conn.close()
	except:
		print('unable to execute query')
		print(sys.exc_info()[0])

File: gemini/js/db_libs.js
const bid_ask_create = (symbol) => `
CREATE TABLE gemini_${symbol}_bid_ask (
	ts			timestamp without time zone,
	symbol		varchar(6),
	type		boolean,
	price		real,
	reamining real
)
;
`


const trade_create = (symbol) => `
CREATE TABLE gemini_${symbol}_trade (
	ts 			timestamp without time zone,
	symbol		varchar(6),
	type		boolean,
	price		real,
	amount		real
)
;
`


const bid_ask_drop = (symbol) => `
DROP TABLE gemini_${symbol}_bid_ask
;
`


const trade_drop = (symbol) => `
DROP TABLE gemini_${symbol}_trade
;
`


const bid_ask_insert = (symbol, type, price, remaining) => `
INSERT INTO gemini_${symbol}_bid_ask (
	ts,
	symbol,
	type,
	price,
	remaining
) VALUES (
	CURRENT_TIMESTAMP,
	'${symbol}',
	${type},
	${price},
	${remaining}
)
;
`


const trade_insert = (symbol, ts, type, price, amount) => `
INSERT INTO gemini_${symbol}_trade (
	ts,
	symbol,
	type,
	price,
	amount
) VALUES (
	TO_TIMESTAMP(${ts}::double precision / 1000),
	'${symbol}',
	${type},
	${price},
	${amount}
)
;
`


module.exports = {
	bid_ask_create: bid_ask_create,
	trade_create: trade_create,
	bid_ask_drop: bid_ask_drop,
	trade_drop: trade_drop,
	bid_ask_insert: bid_ask_insert,
	trade_insert: trade_insert
}